src/controllers/v1/attachment/index.ts

```diff
@@ -204,10 +204,25 @@ export class AttachmentController extends Controller {
   @Produces('application/json')
   @Produces('application/octet-stream')
   @SuccessResponse(200)
-  public async getById(@Request() req: express.Request, @Path() id: UUID): Promise<unknown | Readable> {
-    this.log.debug(`attempting to retrieve ${id} attachment`)
-    const [attachment] = await this.db.get('attachment', { id })
-    if (!attachment) throw new NotFound('attachment')
+  public async getById(@Request() req: express.Request, @Path() id: UUID | string): Promise<unknown | Readable> {
+    this.log.debug(`attempting to retrieve attachment with id or hash: ${id}`)
+    let attachment: AttachmentRow | undefined = undefined
```

Comment:
instead of making this mutable a better strategy would be to set the where clause based on a ternary on `isValidUUID(id)`. That way you can make it `const` which is preferable. (It also gets rid of the evil `;`